src/components/UserSelector.tsx

```diff
@@ -1,45 +1,34 @@
-import React, { useContext, useEffect, useState } from 'react';
+import { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
-import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import * as authorActions from '../features/author';
+import * as usersActions from '../features/users';
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
-
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
-  // `users` are loaded from the API, so for the performance reasons
-  // we load them once in the `UsersContext` when the `App` is opened
-  // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+export const UserSelector = () => {
   const [expanded, setExpanded] = useState(false);
+  const dispatch = useAppDispatch();
+  const { users } = useAppSelector(state => state.users);
+  const { author } = useAppSelector(state => state.author);
+
+  useEffect(() => {
+    dispatch(usersActions.initUsers());
```

Comment:
The `initUsers` action should be dispatched only if it is necessary to initialize the users list. Ensure that this action is not dispatched multiple times unnecessarily.